feat: object store to pvc migrations (#633)
replicatedhq/kurl.sh: src/markdown-pages/add-ons/registry.md

## Registry Storage Backends 

When installed as part of a kURL spec that contains an object store (e.g., Rook or MinIO), the Registry add-on will use that API as a storage backend.
If an object store is not available, a Persistent Volume (PV) will be used as the storage backend.

> Migration can be performed on upgrade/re-install from an object store to a PV under the following circumstances.

Reviewer: ```suggestion
Migration can be performed on upgrade/re-install from an object store to a PV under the following circumstances:
```

> Migration can be performed on upgrade/re-install from an object store to a PV under the following circumstances.

Reviewer: ```suggestion
Migrations are performed on upgrade/re-install from an object store to a PV under the following circumstances.
```
